app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java

```diff
@@ -0,0 +1,177 @@
+import android.graphics.Color;
+import android.location.LocationManager;
+import android.net.ConnectivityManager;
+import android.net.wifi.ScanResult;
+import android.net.wifi.WifiInfo;
+import android.net.wifi.WifiManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.android.splitpersonality.R;
+import com.google.android.gms.location.LocationRequest;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class WifiActivity extends AppCompatActivity {
+
+    WifiManager manager;
+    WifiReceiver receiver;
+    LocationManager locationManager;
+    TextView text;
+    List<ScanResult> list;
+    ArrayList<String> provider,x,y;
+    recyclerAdapter adapter;
+    ListView listView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_wifi);
+        manager = (WifiManager) this.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
+        provider = new ArrayList<>();
+        x = new ArrayList<>();
+        y = new ArrayList<>();
+        listView = findViewById(R.id.list);
+        text=findViewById(R.id.text);
+        locationManager = (LocationManager) this.getSystemService(Context.LOCATION_SERVICE);
+        location();
+    }
+
+    public void location()
+    {
+        if( !locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER) ) {
+            new AlertDialog.Builder(this)
+                    .setTitle("Location")
+                    .setMessage("Please enable location")
+                    .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialogInterface, int i) {
+                            startActivity(new Intent(android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS));
+                        }
+                    })
+                    .setNegativeButton("No", new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            finish();
+                        }}
+                    )
+                    .show();
+        }
+    }
+
+    //start scan
+    public void scanning()
+    {
+        receiver = new WifiReceiver();
+        registerReceiver(receiver, new IntentFilter(manager.SCAN_RESULTS_AVAILABLE_ACTION));
+        manager.startScan();
+    }
+    //display keywords based on DBm strength
+    public String returnlevel(int DBm)
+    {
+        if(DBm >= -50) return DBm+" dBm > Excellent";
+        else if (DBm < -50 && DBm >= -60) return DBm+" dBm > Good";
+        else if (DBm < -60 && DBm >= -70) return DBm+" dBm > Fair";
```

Comment:
Remove unnecessary conditions

Reply:
This condition is required so as to show the strength of wifi to the user.